packages/slice-machine/src/modules/slices/index.ts

```diff
@@ -80,15 +98,48 @@ export function* createSliceSaga({
   }/${variationId}`;
 }
 
+export function* renameSliceSaga({
+  payload,
+}: ReturnType<typeof renameSliceCreator.request>) {
+  try {
+    yield call(
+      renameSlice,
+      payload.sliceId,
+      payload.newSliceName,
+      payload.libName
+    );
+    window.location.href = `/${payload.libName.replace(/\//g, "--")}/${
```

Comment:
would have preferred to use `next/router` to keep everything working as an SPA should.
I guess that'll have to be a tech debt ticket

Reply:
Maybe we can get on a call to tackle this together? I tried this and it came up with a bunch of errors.